pkg/instancegroups/instancegroups.go

```diff
@@ -248,6 +248,20 @@ func (r *RollingUpdateInstanceGroup) validateClusterWithDuration(rollingUpdateDa
 	}
 }
 
+// validateClusterWithDurationLoop validates cluster 3 times to make sure cluster does not flap
+func (r *RollingUpdateInstanceGroup) validateClusterWithDurationLoop(rollingUpdateData *RollingUpdateCluster, duration time.Duration) error {
+	var err error
+	for i := 1; i <= 3; i++ {
+		err = r.validateClusterWithDuration(rollingUpdateData, duration)
```

Comment:
I think this is the wrong way around—this verify-the-success-holds loop should be called from within `validateClusterWithDuration()`. We don't want three successful validations with validation failures between them, We want some number of successful validations in a row. If a successful validation is followed by a failed validation, it doesn't count.

Reply:
I do not now understand, this is what it does? This makes sure that we have 3 validations in a row. If there is fails in any of them it will fail. Please remember that validateClusterWithDuration checks the cluster many times.

Reply:
If the sequence is fail, succeed, fail, succeed, fail, succeed, then `validateClusterWithDurationLoop()` will incorrectly return.

Reply:
its not incorrect? If will return failed in that case. 

We must remember that if we receive single fail from `validateClusterWithDuration` it means that cluster has been already in incorrect state for `duration` amount of time. Like in our case that means that it did not validate in 20 minutes. In my opinion in that case we should not even go to next loop, instead exit with failure state.

Thing what I am trying to achieve here is following:

succeed (loop `duration` amount of time to achieve that), succeed (loop `duration` amount of time to achieve that).

In case of flapping the first succeed is achieved somewhere between 5-15minutes. The next succeed should be pretty fast in our case, because its just controller manager which is pending state and will start in minute or so.

Reply:
I'm saying if the `ClusterValidator.Validate()` returns that sequence of statuses, then `validateClusterWithDurationLoop()` will incorrectly return.

Do we really want to bake in an assumption about the number of times it is going to flap? It just seems to me to be more straightforward to do the success looping from inside `tryValidateCluster()`.